core/consensus/consensus_event.hpp

```diff
@@ -27,47 +27,104 @@ limitations under the License.
 
 #include "../crypto/signature.hpp"
 #include "../util/logger.hpp"
-#include "../model/transactions/abstract_transaction.hpp"
+#include "../model/transaction.hpp"
 
-namespace consensus_event {
-
-struct ConsensusEvent {
+#include "../service/json_parse.hpp"
 
-    std::unique_ptr<abstract_transaction::AbstractTransaction> tx;
-    std::unordered_map<std::string, std::string> txSignatures; // map of public keys→signatures
-
-    std::string merkleRootHash;
-    std::unordered_map<std::string, std::string> merkleRootSignatures;
+namespace consensus_event {
 
+template<
+        typename T, typename U,
+        std::enable_if_t<
+                std::is_base_of<transaction::Transaction<U>, T>::value,std::nullptr_t
+        > = nullptr
+>
+class ConsensusEvent {
+
+
+    struct eventSignature{
+        std::string publicKey;
+        std::string signature;
+
+        eventSignature(
+            std::string pubKey,
+            std::string sig
+        ):
+                publicKey(pubKey),
+                signature(sig)
+        {}
+    };
+    T tx;
+    std::vector<eventSignature> eventSignatures;
+
+public:
     unsigned long long order = 0;
 
-    ConsensusEvent(std::unique_ptr<abstract_transaction::AbstractTransaction> atx):
-        tx(std::move(atx))
+    ConsensusEvent(T atx):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class 'ConsensusEvent' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4345543327/issues/source?bid=3712491&fileBranchId=3870389#l63)